trail/query/views.py

```diff
@@ -61,3 +66,162 @@ def print_results(request):
 
     return render(request, "query.html",
                   {"data": query_results, "wcsdata": wcs_list, "queryform": form, "render_table": True})
+
+
+class MetadataDAO(APIView):
+    """
+     Data Accessing objects for metadata, all query for metadatas are implemented here
+
+    Notes
+    -----
+    """
+
+    RALOW = "raLow"
+    RAHIGH = "raHigh"
+    DECLOW = "decLow"
+    DECHIGH = "decHigh"
+
+    """Support Query that returns metadata entry as a result through metadata info
+
+    Notes
+    -----
+    """
+    # gets all the metadata entry that matches the query parameters.
+    def getMetadataByParams(self, paramDict):
+
+        """
+        Returns the Metadata entries that meets the users input
+        Parameters
+        ----------
+        fields: list of wanted field in the metadata returned
+        queryParam: list of criterias for metadata
+
+        """
+
+        querySet = Metadata.objects.all()
+        if paramDict is not None:
+            querySet = querySet.filter(**paramDict)
+
+        return list(querySet)
+
+    def getMeatadataInSpecifiedSky(self, paramDict):
+
+        """
+        Returns the Metadata associated with the part of sky the user specifies
+        ----------
+
+        Parameters
+        ----------------
+        raLow: the lower bound for ra
+        raHigh: the upper bound for ra
+        decLow: the lower bound for dec
+        decHigh: the upper bound for dec
+
+        Note
+        -------------------
+        if the parameters is not specified correctly, will return a bad request
+
+        """
+        if self.isWcsQueryParamMissing(paramDict):
+            return []
+
+        lowerRight = self.getXYZFromWcs(float(paramDict.get(self.RALOW)),
+                                        float(paramDict.get(self.DECLOW)))
+        upperLeft = self.getXYZFromWcs(float(paramDict.get(self.RAHIGH)),
+                                       float(paramDict.get(self.DECHIGH)))
+
+        filteredWcs = Wcs.objects.all().filter(**self.makeFilterDictForWcs(upperLeft, lowerRight))
+        # for every wcs, find the metadata id
+        metadataIds = set([metadataId['metadata'] for metadataId in filteredWcs.values('metadata')])
+        metadatas = self.getMetadatasByIds(metadataIds)
+
+        return metadatas
+
+    def getMetadatasByIds(self, metadataIds):
```

Comment:
These and the few below are tiny functions, but I would still appreciate one-liner docstrings for each